src/modules/network/components/NetworkSwitch.vue

```diff
@@ -160,28 +149,42 @@ export default {
      * @returns {string[]}
      */
     typeNames() {
-      const unsorted = Object.keys(types);
-      unsorted.splice(unsorted.indexOf('ETH'), 1);
-      unsorted.sort();
-      const test = unsorted.filter(item => {
-        return types[item].isTestNetwork;
-      });
-      const main = unsorted.filter(item => {
-        return !types[item].isTestNetwork;
-      });
-      const sorted = main.concat(test);
-      sorted.unshift('ETH');
-      return sorted;
+      if (this.hasNetworks) {
+        const unsorted =
+          this.filterTypes.length > 0
+            ? [...this.filterTypes]
+            : Object.keys(types);
+        unsorted.splice(unsorted.indexOf('ETH'), 1);
+        unsorted.sort();
+        const test = unsorted.filter(item => {
+          return types[item].isTestNetwork;
+        });
+        const main = unsorted.filter(item => {
+          return !types[item].isTestNetwork;
+        });
+        const sorted = main.concat(test);
+        sorted.unshift('ETH');
```

Comment:
what if `ETH` is already part of the list?

Reply:
it's just moving it to front

Reply:
So it always in the front even after sorting